src/persist-client/src/fetch.rs

```diff
@@ -701,13 +702,6 @@ where
                 inline_desc,
                 registered_desc
             );
-            assert!(
```

Comment:
This had to be removed because a rewritten batch can have a registered (HollowBatch) desc that is arbitrarily far ahead of the one written to s3. We could keep this check for any cases where the rewrite feature has not been used.... except that we can't currently distinguish between "not used" vs "called with minimum timestamp". That could be fixed by switching the ts_rewrite to be an Option, but I discovered this at the last minute, so figured I'd check in again about how we feel about `None` vs `Antichain::from_elem(T::minimum())` before typing it all up.

Personally, I'm a bit sad to lose this, but not sure I particularly think it's critical.

Reply:
> I'd check in again about how we feel about None vs Antichain::from_elem(T::minimum())

I don't feel particularly strongly! Normally I have a mild preference for the latter (since no rewrite is equivalent to the trivial rewrite in practice) but this is a good reason to care about the difference and I can imagine others.